nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/stream/StreamServiceImpl.java

```diff
@@ -123,40 +129,59 @@ protected Path getChroniclePath(String basePath, String name) {
+    protected void createStreamIfNotExists(LogConfigDescriptor config) {
+        if (config.logs.isEmpty()) {
             return;
         }
-        LogManager manager = getLogManager(name);
-        config.getLogsToCreate().forEach((stream, size) -> {
-            log.info("Create if not exists stream: " + stream + " with manager: " + name);
-            manager.createIfNotExists(stream, size);
+        LogManager manager = getLogManager(config.getId());
+        config.logs.forEach(l -> {
+            getLog().info("Create if not exists stream: " + l.getId() + " with manager: " + config.getId());
+            manager.createIfNotExists(l.getId(), l.size);
         });
     }
 
     @Override
     public void start(ComponentContext context) {
         super.start(context);
-        configs.forEach(this::createStreamIfNotExists);
-        processorDescriptors.forEach(this::initProcessor);
+        List<LogConfigDescriptor> logDescs = getDescriptors(XP_LOG_CONFIG);
+        logDescs.forEach(this::createStreamIfNotExists);
+        List<StreamProcessorDescriptor> streamDescs = getDescriptors(XP_STREAM_PROCESSOR);
+        streamDescs.forEach(this::initProcessor);
         new ComponentsLifeCycleListener().install();
     }
 
-    protected void initProcessor(String name, StreamProcessorDescriptor descriptor) {
-        if (processors.containsKey(name)) {
-            log.error("Processor already initialized: " + name);
+    protected void initProcessor(StreamProcessorDescriptor descriptor) {
+        if (processors.containsKey(descriptor.getId())) {
+            getLog().error("Processor already initialized: " + descriptor.getId());
             return;
         }
-        log.info("Init Stream processor: " + name + " with manager: " + descriptor.config);
+        getLog().info("Init Stream processor: " + descriptor.getId() + " with manager: " + descriptor.config);
         LogManager manager = getLogManager(descriptor.config);
-        Topology topology = descriptor.getTopology();
-        StreamProcessor streamProcessor = new LogStreamProcessor(manager);
-        Settings settings = descriptor.getSettings(Framework.getService(CodecService.class));
-        if (log.isDebugEnabled()) {
-            log.debug("Starting computation topology: " + name + "\n" + topology.toPlantuml(settings));
+        try {
+            Topology topology = descriptor.klass.getDeclaredConstructor().newInstance().getTopology(descriptor.options);
+            StreamProcessor streamProcessor = new LogStreamProcessor(manager);
+            Settings settings = getSettings(descriptor);
+            if (getLog().isDebugEnabled()) {
+                getLog().debug(
+                        "Starting computation topology: " + descriptor.getId() + "\n" + topology.toPlantuml(settings));
+            }
+            streamProcessor.init(topology, settings);
+            processors.put(descriptor.getId(), streamProcessor);
+        } catch (ReflectiveOperationException e) {
```

Comment:
You should restrict the scope of the try/catch to just the `getDeclaredConstructor().newInstance()`